components/Modals/UpdateCharacterMetadataModal.tsx

```diff
@@ -156,12 +156,24 @@ export const UpdateCharacterMetadataModal: React.FC = () => {
 
         setIsUpdating(true);
 
-        const characterMetadata = {
+        const characterMetadata: {
+          name: string;
+          description: string;
+          image: string;
+          attributes?: {
+            trait_type: string;
+            value: string;
+          }[];
+        } = {
           name: newName,
           description: newDescription,
           image: `ipfs://${cid}`,
         };
 
+        if (!newAvatarFile && selectedCharacter.attributes) {
+          characterMetadata['attributes'] = selectedCharacter.attributes;
+        }
+
```

Comment:
And we also need to update the base URI on the contract to our server endpoint. but it would fail the old metadata api calls. So we may want to setup an api endpoint `https://characters.io/[ipfsHash]` or something. 

Needs thinking. Wdyt @ECWireless  ?

Reply:
oh ya, good point. What about `https://characters.io/ipfs/[ipfsHash]` on `prod` and `https://dev.characters.io/ipfs/[ipfsHash]` on dev, since the databases will be different?

Reply:
Switched metadata endpoint in bc222d88faac902a2fc87339daea03e04d672571